diff --git a/src/Plugin/facets/facet_source/SearchApiDisplay.php b/src/Plugin/facets/facet_source/SearchApiDisplay.php
index 57911d5..a167e95 100644
--- a/src/Plugin/facets/facet_source/SearchApiDisplay.php
+++ b/src/Plugin/facets/facet_source/SearchApiDisplay.php
@@ -117,7 +117,7 @@ class SearchApiDisplay extends FacetSourcePluginBase implements SearchApiFacetSo
       $container->get('plugin.manager.search_api.display'),
       // Support 9.3+.
       // @todo remove switch after 9.3 or greater is required.
-      version_compare(\Drupal::VERSION, '9.3', '>=') ? $request_stack->getMainRequest() : $request_stack->getMasterRequest(),
+      version_compare(\Drupal::VERSION, '9.3', '>=') ? $request_stack->getMasterRequest() : $request_stack->getMainRequest(),
       $container->get('module_handler')
     );
   }
